src/inc/regdisp.h

```diff
@@ -72,7 +71,71 @@ struct REGDISPLAY : public REGDISPLAY_BASE {
     DWORD * pEax;
 
     DWORD * pEbp;
+#endif // !WIN64EXCEPTIONS
+
+#ifndef WIN64EXCEPTIONS
+
+#define VOLATILE_REG_METHODS(reg) \
+    inline PDWORD Get##reg##Location(void) { return p##reg;  } \
+    inline void   Set##reg##Location(PDWORD p##reg) { this->p##reg = p##reg; }
+
+#define NONVOLATILE_REG_METHODS(reg) VOLATILE_REG_METHODS(reg)
+
+#else // !WIN64EXCEPTIONS
+
+#define VOLATILE_REG_METHODS(reg) \
+    inline PDWORD Get##reg##Location(void) { return &pCurrentContext->reg; } \
+    inline void   Set##reg##Location(PDWORD p##reg) { pCurrentContext->reg = *p##reg; }
```

Comment:
The pCurrentContext->XXX contain values that the pCurrentContextPointers->XXX point to. Please note that there are actually two context and two context pointer structures. To one of them points the pCurrentContext / pCurrentContextPointers. To the other points the pParentContext / pParentContextPointers. That allows unwinding at some places to eliminate some context / context pointers copying by just switching the the pCurrentXXX and pParentXXX pointers.

While it might theoretically be possible to modify the existing code to not to use pCurrentContext for x86, it would be error prone and make the x86 Linux code even more different from the other targets using WIN64EXCEPTIONS.